File: src/main/java/com/kenvix/utils/lang/CallerClass.java
//--------------------------------------------------
// Class CallerClass
//--------------------------------------------------
// Written by Kenvix <i@com.kenvix.com>
//--------------------------------------------------

package com.kenvix.utils.lang;

import org.jetbrains.annotations.NotNull;

public class CallerClass extends SecurityManager {
    private static final CallerClass INSTANCE = new CallerClass();

    private CallerClass() {
    }

    @NotNull
    public static Class[] get() {
        return INSTANCE.getClassContext();
    }

    @NotNull
    public static Class get(int depth) {
        return INSTANCE.getClassContext()[depth];
    }
}

File: src/main/java/com/kenvix/utils/lang/CallerClass.kt
//--------------------------------------------------
// Class CallerClass
//--------------------------------------------------
// Written by Kenvix <i@com.kenvix.com>
//--------------------------------------------------
package com.kenvix.utils.lang

@Suppress("unused")
object CallerClass : SecurityManager() {
    fun get(): Array<Class<*>> {
        return classContext
    }

    operator fun get(depth: Int): Class<*> {
        return classContext[depth]
    }

    public override fun getClassContext(): Array<Class<*>> = super.getClassContext()
}